app/src/main/java/com/android/streetworkapp/ui/train/TrainParamScreen.kt

```diff
@@ -0,0 +1,367 @@
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material3.Button
+import androidx.compose.material3.ButtonDefaults
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableIntStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.runtime.snapshotFlow
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.platform.testTag
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import androidx.compose.ui.unit.times
+import com.android.streetworkapp.ui.navigation.NavigationActions
+import com.android.streetworkapp.ui.theme.ColorPalette.INTERACTION_COLOR_DARK
+import com.android.streetworkapp.ui.theme.ColorPalette.PRIMARY_TEXT_COLOR
+import com.android.streetworkapp.ui.theme.ColorPalette.PRINCIPLE_BACKGROUND_COLOR
+
+@SuppressLint("DefaultLocale")
+@Composable
+fun TrainParamScreen(
+    navigationActions: NavigationActions,
+    activity: String,
+    isTimeDependent: Boolean,
+    type: String,
+    paddingValues: PaddingValues = PaddingValues(0.dp)
+) {
+  var minutes by remember { mutableIntStateOf(0) }
+  var seconds by remember { mutableIntStateOf(0) }
+  var sets by remember { mutableIntStateOf(1) }
+  var reps by remember { mutableIntStateOf(10) }
+
+  Column(
+      modifier =
+          Modifier.fillMaxSize().padding(paddingValues).padding(16.dp).testTag("TrainParamScreen"),
+      horizontalAlignment = Alignment.CenterHorizontally,
+      verticalArrangement = Arrangement.Center) {
+        TrainParamHeader(
+            isTimeDependent = isTimeDependent,
+            minutes = minutes,
+            seconds = seconds,
+            sets = sets,
+            reps = reps,
+            activity = activity)
+
+        if (isTimeDependent) {
+          TimerInputGrid(
+              minutes = minutes,
+              seconds = seconds,
+              onUpdateMinutes = { minutes = it },
+              onUpdateSeconds = { seconds = it })
+        } else {
+          SetsAndRepsSection(
+              sets = sets, reps = reps, onUpdateSets = { sets = it }, onUpdateReps = { reps = it })
+        }
+
+        ConfirmActionButton(
+            navigationActions = navigationActions,
+            activity = activity,
+            isTimeDependent = isTimeDependent,
+            type = type,
+            minutes = minutes,
+            seconds = seconds,
+            sets = sets,
+            reps = reps)
+      }
+}
+
+@SuppressLint("DefaultLocale")
+@Composable
+fun TrainParamHeader(
+    isTimeDependent: Boolean,
+    minutes: Int,
+    seconds: Int,
+    sets: Int,
+    reps: Int,
+    activity: String
```

Comment:
I believe this approach will increase future implementation time. Instead of emphasizing argument precision, I suggest placing all arguments in the correct order. This improves readability without adding much boilerplate.
What do you think ?

Reply:
If you think you will have to heavily change arguments later, i can understand it